src/pages/permissions/panels/EditIdpGroupPanel.tsx

```diff
@@ -0,0 +1,369 @@
+          role: "cell",
+          "aria-label": "Modified status",
+          className: "modified-status u-align--right",
+        },
+      ],
+      sortData: {
+        name: group.name.toLowerCase(),
+        description: group.description.toLowerCase(),
+      },
+    };
+  });
+
+  const groupsTable = groups.length ? (
+    <ScrollableTable
+      dependencies={[
+        groups,
+        modifiedGroups.size,
+        searchGroup,
+        notify.notification,
+        formik,
+      ]}
+      tableId="idp-group-mapping-table"
+      belowIds={["panel-footer"]}
+    >
+      <SelectableMainTable
+        id="idp-group-mapping-table"
+        headers={headers}
+        rows={rows}
+        sortable
+        emptyStateMsg="No IDP groups found"
+        itemName="IDP group"
+        parentName=""
+        selectedNames={Array.from(selectedGroups)}
+        setSelectedNames={modifyGroups}
+        processingNames={[]}
+        filteredNames={groups.map((group) => group.name)}
+        onToggleRow={toggleRow}
+        hideContextualMenu
+      />
+    </ScrollableTable>
+  ) : (
+    <EmptyState
+      className="empty-state full-width show-left-icon"
+      image={<Icon name="plans" className="empty-state-icon" />}
+      title="No groups"
+    >
+      <p className="u-no-margin--bottom">There are no groups in the system.</p>
+      <Link to={`/ui/permissions/groups`}>
+        Create groups
+        <Icon className="external-link-icon" name="external-link" />
+      </Link>
+    </EmptyState>
+  );
+
+  const nameModified = !!formik.touched.name;
+  const nameIsValid = formik.isValid && formik.values.name;
+  const groupsModified = !!modifiedGroups.size;
+  const enableSubmission =
+    (nameModified && nameIsValid) || (nameIsValid && groupsModified);
+
+  return (
+    <>
+      <SidePanel
+        isOverlay
+        loading={isLoading}
+        hasError={!groups}
+        className="idp-group-edit-panel"
+      >
+        <SidePanel.Header>
+          <SidePanel.HeaderTitle>{`Edit IDP group ${idpGroup?.name}`}</SidePanel.HeaderTitle>
+        </SidePanel.Header>
+        <NotificationRow className="panel-notification" />
+        <SidePanel.Content className="u-no-padding">
+          <Form onSubmit={formik.handleSubmit}>
+            {/* hidden submit to enable enter key in inputs */}
+            <Input type="submit" hidden value="Hidden input" />
+            <Input
+              type="text"
+              label="Name"
+              required
+              autoFocus
+              id="name"
+              name="name"
+              onBlur={formik.handleBlur}
+              onChange={formik.handleChange}
+              value={formik.values.name}
+              error={formik.touched.name ? formik.errors.name : null}
+              placeholder="Enter name"
+            />
+          </Form>
+          <PermissionGroupsFilter
+            onChange={setSearchGroup}
+            value={searchGroup}
+          />
+          {groupsTable}
```

Comment:
Can we extract all of this (and the `groupsTable` code above) to a component that is reused on create and edit?

Reply:
Moved the form `IdpGroupForm.tsx` and the table into `IdpGroupMappingsTable.tsx`.